src/System.Private.CoreLib/shared/System/TimeZoneInfo.Unix.cs

```diff
@@ -500,28 +529,32 @@ private static string FindTimeZoneId(byte[] rawData)
             string posixrulesFilePath = Path.Combine(timeZoneDirectory, "posixrules");
             byte[] buffer = new byte[rawData.Length];
 
-            try
-            {
-                foreach (string filePath in EnumerateFilesRecursively(timeZoneDirectory))
+            bool FindTzFile(string filePath)
+            {                        
+                // skip the localtime and posixrules file, since they won't give us the correct id
+                if (!string.Equals(filePath, localtimeFilePath, StringComparison.OrdinalIgnoreCase)
+                    && !string.Equals(filePath, posixrulesFilePath, StringComparison.OrdinalIgnoreCase))
                 {
-                    // skip the localtime and posixrules file, since they won't give us the correct id
-                    if (!string.Equals(filePath, localtimeFilePath, StringComparison.OrdinalIgnoreCase)
-                        && !string.Equals(filePath, posixrulesFilePath, StringComparison.OrdinalIgnoreCase))
+                    if (CompareTimeZoneFile(filePath, buffer, rawData))
                     {
-                        if (CompareTimeZoneFile(filePath, buffer, rawData))
-                        {
-                            // if all bytes are the same, this must be the right tz file
-                            id = filePath;
+                        // if all bytes are the same, this must be the right tz file
+                        id = filePath;
 
-                            // strip off the root time zone directory
-                            if (id.StartsWith(timeZoneDirectory, StringComparison.Ordinal))
-                            {
-                                id = id.Substring(timeZoneDirectory.Length);
-                            }
-                            break;
+                        // strip off the root time zone directory
+                        if (id.StartsWith(timeZoneDirectory, StringComparison.Ordinal))
+                        {
+                            id = id.Substring(timeZoneDirectory.Length);
                         }
+                        return true;
                     }
                 }
+
+                return false;
+            }
+
+            try
+            {
+                FindFile(timeZoneDirectory, FindTzFile);
```

Comment:
Nit: I think this would be more readable when written as closure.
```
EnumerateFilesRecursively(timeZoneDirectory, (string filePath) =>
{
    // skip the localtime and posixrules file, since they won't give us the correct id
    if (!string.Equals(filePath, localtimeFilePath, StringComparison.OrdinalIgnoreCase)
    ....
});
```

Local functions make sense when they are used multiple times, or to factor our a cleanly separated part of the algorithm from a very large method.